wiki/C++-Style-Guide.md

```diff
@@ -0,0 +1,814 @@
+# Table of Contents
+
+* [Background](#background)
+* [Files](#files)
+* [Namespaces](#namespaces)
+* [Classes](#classes)
+* [Functions](#functions)
+* [Variables](#variables)
+* [Comments](#comments)
+* [Formatting](#formatting)
+* [Best practices](#best-practices)
+* [C++11](#c11)
+
+# Background
+
+This style guide is also available on the [website](https://endless-sky.github.io/styleguide/styleguide.xml).
+
+Each section of this style guide can be read as a "how to": how to create source code files, how to create classes, etc. The sections are ordered hierarchically, from biggest "units" (files, namespaces, classes) to increasingly smaller ones (variables, comments, formatting). Our style choices are based on a small set of principles:
+
+- Readability is achieved by laying out text in a clear visual hierarchy.
+- Code should not rely on features of a particular editor or compiler.
+- Formatting should be consistent to avoid confusing or distracting the reader.
+- Reducing the risk of bugs is important even if it takes more work.
+
+
+# Files
+
+This section deals with naming and encoding of source code files and the items that appear at the start of each file.
+
+## Line endings
+
+All files should have Unix-style line endings.
+That is, "\n" alone, not "\r\n" (Windows style). If an editor changes Unix line endings to Windows style, this causes issues with source control diffs because those lines appear to be changed even though their actual text is the same.
```

Comment:
```suggestion
That is, `\n` alone, not `\r\n` (Windows style). If an editor changes Unix line endings to Windows style, this causes issues with source control diffs because those lines appear to be changed even though their actual text is the same.
```